appium/tests/helpers/TouchHelper.ts

```diff
@@ -29,6 +30,27 @@ class TouchHelper {
       {action: 'release', options: {}},
     ]);
   }
+
+  static scrollDownToElement = async (element: WebdriverIO.Element) => {
+    const { width, height } = await driver.getWindowSize();
+    const anchor = width / 2;
+    const startPoint = height * 0.25;
+    const endPoint = height * 0.15;
+    let index = 0;
+
+    do {
+      await browser.pause(1000); // due to scroll action which takes about second
+      await driver.touchPerform([
+        {action: 'press', options: {x: anchor, y: startPoint}},
+        {action: 'wait', options: {ms: 100}},
+        {action: 'moveTo', options: {x: anchor, y: endPoint}},
+        {action: 'release', options: {}},
+      ]);
+    } while (await (await element).isDisplayed() !== true  && index++ < 14);
```

Comment:
This kind of loop declaration is prone to off-by-one errors. I can't confirm if it's right or not from looking at it, and I think it's probably indeed off by one.

I recommend to do:

```ts
// this wait can be later replaced by waiting for loader to go away before scrolling
await browser.pause(1000); // make sure contents are loaded first, so we don't scroll too early
for(let i = 0; i < 15; i++) {
    if (await (await element).isDisplayed()) {
        return;
    }
    await driver.touchPerform([
      {action: 'press', options: {x: anchor, y: startPoint}},
      {action: 'wait', options: {ms: 100}},
      {action: 'moveTo', options: {x: anchor, y: endPoint}},
      {action: 'release', options: {}},
    ]);
    await browser.pause(1000); // due to scroll action which takes about second
}
throw new Error(`Element ${JSON.stringify(element.selector)} not displayed after scroll`);
```

You should find it easier to read and follow when you structure it that way. And it will more likely be correct.